app/controllers/games_controller.rb

```diff
@@ -13,7 +13,12 @@ def create
 
   def show
     @game = Game.find_by_id(params[:id])
-    render text: 'Not Found', status: :not_found if @game.blank?
+    if @game.blank?
+      render text: 'Not Found', status: :not_found
+      return
+    end
+    flash[:notice] = 'Black King Check!' if @game.check?('black')
+    flash[:notice] = 'White King Check!' if @game.check?('white')
```

Comment:
Right now the flash notice stays up after the first move to get out of check. To fix that, you can set it to nil when not in check.

 if @game.check?('black')
     flash[:notice] = 'Black King Check!'
    elsif @game.check?('white')
      flash[:notice] = 'White King Check'
    else
     flash[:notice] = nil
    end